packages/cli/src/license/license.controller.ts

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-param-reassign */
+
+import express from 'express';
+import { LoggerProxy } from 'n8n-workflow';
+
+import { getLogger } from '@/Logger';
+import {
+	ILicensePostResponse,
+	ILicenseReadResponse,
+	InternalHooksManager,
+	ResponseHelper,
+} from '..';
+import { LicenseService } from './License.service';
+import { getLicense } from '@/License';
+import { LicenseRequest } from '@/requests';
+import { isInstanceOwner } from '@/PublicApi/v1/handlers/users/users.service';
+
+export const licenseController = express.Router();
+
+/**
+ * Initialize Logger if needed
+ */
+licenseController.use((req, res, next) => {
+	try {
+		LoggerProxy.getInstance();
+	} catch (error) {
+		LoggerProxy.init(getLogger());
+	}
+	next();
+});
+
+// Helper for getting the basic license data that we want to return
+async function getLicenseData(): Promise<ILicenseReadResponse> {
+	const triggerCount = await LicenseService.getActiveTriggerCount();
+	const license = getLicense();
+	const mainPlan = license.getMainPlan();
+
+	return {
+		usage: {
+			executions: {
+				value: triggerCount,
+				limit: license.getTriggerLimit(),
+				warningThreshold: 0.8,
+			},
+		},
+		license: {
+			planId: mainPlan?.productId ?? '',
+			planName: license.getPlanName(),
+		},
+	};
+}
+
+/**
+ * GET /license
+ * Get the license data, usable by everyone
+ */
+licenseController.get(
+	'/',
+	ResponseHelper.send(async (): Promise<ILicenseReadResponse> => {
+		return getLicenseData();
+	}),
+);
+
+/**
+ * POST /license/activate
+ * Only usable by the instance owner, activates a license.
+ */
+licenseController.post(
+	'/activate',
+	ResponseHelper.send(async (req: LicenseRequest.Activate): Promise<ILicensePostResponse> => {
+		// First ensure that the requesting user is the instance owner
+		if (!isInstanceOwner(req.user)) {
+			LoggerProxy.info('Non-owner attempted to activate a license', {
+				userId: req.user.id,
+			});
+			throw new ResponseHelper.NotFoundError('Only an instance owner may activate a license');
+		}
```

Comment:
I also recommend moving these checks to a middleware - also I think we should return forbidden (403) instead of not found (404)

Reply:
well there's one view in this endpoint that doesn't require this check, wouldn't that break if I moved this there?

Reply:
We can have the middleware check the route and only block non-owner users when it's `/renew` or `/activate`